subprojects/luajit/src/host/meson.build

```diff
@@ -11,25 +11,33 @@ endif
 dynasm_dasc = files('../vm_@0@.dasc'.format(dynasm_arch))
 dasm = [minilua, files('../../dynasm/dynasm.lua')]
 
-# BUG: meson does not resolve paths correctly for subprojects
-hpre = '#include "@0@/../lj_arch.h"'.format(meson.current_source_dir())
-
-checkdefs = [
-    ['LJ_ARCH_BITS', '64', ['-D', 'P64']],
-    ['LJ_HASJIT', '1', ['-D', 'JIT']],
-    ['LJ_HASFFI', '1', ['-D', 'FFI']],
-    ['LJ_DUALNUM', '1', ['-D', 'DUALNUM']],
-]
-
-if host_machine.cpu_family() == 'x86'
-    checkdefs += ['__SSE2__', '1', ['-D', 'SSE']]
-endif
-
-foreach def: checkdefs
-    if cc.get_define(def[0], prefix: hpre) == def[1]
-        dasm += def[2]
+if cc.get_id() == 'msvc'
+    # the cl.exe preprocessor seemingly removes/expands macros, so hardcode flags
```

Comment:
Not sure I understand why this is necessary.

Reply:
`get_define` seems to fail unconditionally on MSVC, and as I mentioned in the comment I assume this is because the preprocessor doesn't actually leave the defines intact.